UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -128,10 +132,32 @@ public async Task<List<ShowDailyFilesResponseModel>> GetDailyBatchDetailsFiles(s
                 _logger.LogError(EventIds.ShowDailyFilesResponseFailed.ToEventId(), "Maritime safety information request to get daily NM files failed to return data with exception:{exceptionMessage} for _X-Correlation-ID:{CorrelationId}", ex.Message, correlationId);
                 throw;
             }
-
             return showDailyFilesResponses;
-
         }
 
+        public async Task<ShowWeeklyFilesResponseModel> GetWeeklyFilesResponseModelsAsync(int year, int week, string correlationId)
+        {
+            _logger.LogInformation(EventIds.GetWeeklyFilesResponseStarted.ToEventId(), "Maritime safety information request to get weekly NM files response started with _X-Correlation-ID:{correlationId}", correlationId);
+            ShowWeeklyFilesResponseModel showWeeklyFilesResponses = new();
+            try
+            {
+                showWeeklyFilesResponses.YearAndWeek = await GetAllYearWeek(correlationId);
+                if (year == 0 && week == 0)
+                {
+                    _logger.LogInformation(EventIds.GetWeeklyFilesResponseForYearAndWeekWithZero.ToEventId(), "Maritime safety information request to get weekly NM files response for year and week with zero with _X-Correlation-ID:{correlationId}", correlationId);
```

Comment:
Remove this log as its not adding any value.

Reply:
removed GetWeeklyFilesResponseForYearAndWeekWithZero log